packages/medusa/src/services/totals.ts

```diff
@@ -210,23 +210,42 @@ class TotalsService extends TransactionBaseService {
     }
 
     if (opts.include_tax) {
-      if (isOrder(cartOrOrder) && cartOrOrder.tax_rate !== null) {
-        totals.original_tax_total = Math.round(
-          totals.price * (cartOrOrder.tax_rate / 100)
-        )
-        totals.tax_total = Math.round(
-          totals.price * (cartOrOrder.tax_rate / 100)
-        )
+      if (
+        isOrder(cartOrOrder) &&
+        isDefined(cartOrOrder.tax_rate) &&
+        cartOrOrder.tax_rate !== null
+      ) {
+        const taxRate = cartOrOrder.tax_rate / 100
+
+        const includesTax =
+          this.featureFlagRouter_.isFeatureEnabled(
+            TaxInclusivePricingFeatureFlag.key
+          ) && shippingMethod.includes_tax
+        const taxIncludedInPrice = !shippingMethod.includes_tax
```

Comment:
@srindom is there any scenario where the old `tax_rate` and tax inclusive price can coexist?

Reply:
Generally, no. Could see one (super) edge case where:
- merchant has order (Order X) in their dashboard from old tax system
- merchant updates to tax-inclusive pricing version and creates a shipping option that has tax inclusive price
- merchant updates Order X with the newly created shipping option

That said, we don't have support for updating shipping options on orders through the Admin dashboard. AND the old tax system is so old that any orders from that version will (should) have been processed at this point.

I think it is okay not to address this edge case (i.e. revert the highlighted change here), but don't have strong opinions.